File: tiny-pro/tinypro-menu-service/src/main/java/com/huaweicloud/tinypromenu/CreateMenuDTO.java
package com.huaweicloud.tinypromenu;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.Optional;

@Data
@Getter
@Setter
public class CreateMenuDTO {
    @NotBlank
    @NotNull
    @PositiveOrZero
    public Integer order;
    @NotBlank
    @NotNull
    public String menuType;
    @NotBlank
    @NotNull
    public String name;
    @NotBlank
    @NotNull
    public String path;
    @NotBlank
    @NotNull
    public String component;
    @NotBlank
    @NotNull
    public String icon;
    public Optional<Integer> parentId;
}

File: egg/app/dto/create-menu.dto.ts
const createMenu = {
  order: 'int',
  menuType: 'string',
  name: 'string',
  path: 'string',
  component: 'string',
  icon: 'string',
  parentId: 'int',
} as const;

export type CreateMenu = {
  order: number;
  menuType: string;
  name: string;
  component: string;
  icon: string;
  parentId: number;
};

export {
  createMenu,
};
